src/Microsoft.ML.Dnn/DnnUtils.cs

```diff
@@ -89,6 +92,35 @@ internal static Session LoadTFSession(IExceptionContext ectx, byte[] modelBytes,
             }
             return new Session(graph);
         }
+        internal static void DownloadIfNeeded(Uri address, string fileName)
+        {
+            using (HttpClient client = new HttpClient())
+            {
+                if (File.Exists(fileName))
+                {
+                    var headerResponse = client.GetAsync(address,HttpCompletionOption.ResponseHeadersRead).Result;
+                    var totalSizeInBytes = headerResponse.Content.Headers.ContentLength;
+                    var currentSize = new FileInfo(fileName).Length;
+
+                    //If current file size is not equal to expected file size, re-download file
+                    if (currentSize != totalSizeInBytes)
+                    {
+                        File.Delete(fileName);
+                        var response = client.GetAsync(address).Result;
+                        using FileStream fileStream = new FileStream(fileName, FileMode.Create, FileAccess.Write, FileShare.None);
+                            using Stream contentStream = response.Content.ReadAsStreamAsync().Result;
```

Comment:
>using Stream contentStream = response.Content.ReadAsStreamAsync().Result; [](start = 28, length = 73)

align this line with 110, this is the pattern we follow, this isn't some loop :)